service/history/workflowExecutionContext.go

```diff
@@ -46,15 +46,20 @@ const (
 
 type (
 	workflowExecutionContext interface {
-		appendHistoryEvents(history []*workflow.HistoryEvent, transactionID int64, doLastEventValidation bool) (int, error)
+		appendFirstBatchHistoryForContinueAsNew(newStateBuilder mutableState, transactionID int64) error
+		appendFirstBatchEventsForActive(msBuilder mutableState) (int, error)
+		appendFirstBatchEventsForStandby(msBuilder mutableState, history []*workflow.HistoryEvent) (int, error)
```

Comment:
do we really need to expose 2 appendFirstBatchEvents 1 for active 1 for standby? looks like standby is only used in active

Reply:
passive actually do use it, in here: https://github.com/uber/cadence/pull/1780/files#diff-3f1276f2a53b19aa2890fb159f550a2dR687